applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -1032,7 +1033,7 @@ static void uart_callback(const struct device *dev, struct uart_event *evt, void
 		break;
 	case UART_RX_DISABLED:
 		LOG_DBG("RX_DISABLED");
-		k_sem_give(&rx_change);
+		uart_state = STATE_DISABLED;
```

Comment:
Requires atomic?
Last time I used the `datamode_rx_disabled` without atomic and result not good.

Reply:
Based on tests failing in CI and @junqingzou:s comments. Changing approach.